doc-content/drools-docs/src/main/asciidoc/DecisionEngine/rule-base-configuration-con.adoc

```diff
@@ -0,0 +1,70 @@
+[id='rule-base-configuration-con_{context}']
+
+= Rule base configuration
+
+{PRODUCT} contains a `RuleBaseConfiguration.java` object that you can use to configure exception handler settings, multithreaded execution, and sequential mode in the {DECISION_ENGINE}.
+
+For the rule base configuration options,
+ifdef::DM,PAM[]
+download the *{PRODUCT} [VERSION] Source Distribution* ZIP file from the https://access.redhat.com/jbossnetwork/restricted/listSoftware.html[Red Hat Customer Portal] and navigate to `~/{PRODUCT_FILE}-sources/src/drools-$VERSION/drools-core/src/main/java/org/drools/core/RuleBaseConfiguration.java`.
+endif::[]
+ifdef::DROOLS,JBPM,OP[]
+see the {PRODUCT} https://github.com/kiegroup/drools/blob/7.11.x/drools-core/src/main/java/org/drools/core/RuleBaseConfiguration.java[RuleBaseConfiguration.java] page on GitHub.
+endif::[]
+
+The following rule base configuration options are available for the {DECISION_ENGINE}:
+
+drools.consequenceExceptionHandler::
+When configured, this property defines the class that manages the exceptions thrown by rule consequences. You can use this property to specify a custom exception handler for rule evaluation in the {DECISION_ENGINE}.
++
+--
+Default value: `org.drools.core.runtime.rule.impl.DefaultConsequenceExceptionHandler`
+
+[source]
+----
+drools.consequenceExceptionHandler=org.drools.core.runtime.rule.impl.MyCustomConsequenceExceptionHandler
```

Comment:
I don't follow. This is the system property. Do you mean also set with `KieBaseConfiguration`? Like this?

```
KieServices ks = KieServices.Factory.get();
KieBaseConfiguration kieBaseConf = ks.newKieBaseConfiguration();
kieBaseConf.setOption(MultithreadEvaluationOption.YES);
KieBase kieBase = kieContainer.newKieBase(kieBaseConf);
```

I don't know what that would look like for this property. Please advise.

Reply:
Sorry I misread this part a bit. Please just change this sentence in the description of the property "When configured, this property defines the class that manages the exceptions thrown by rule consequences." to "When configured, this system property defines the class that manages the exceptions thrown by rule consequences."

I just want to be 100% clear that it is a system property.

Reply:
The example could be like this 

KieServices ks = KieServices.Factory.get();
KieBaseConfiguration kieBaseConf = ks.newKieBaseConfiguration();        kieBaseConf.setOption(ConsequenceExceptionHandlerOption.get(MyCustomConsequenceExceptionHandler.class));
KieBase kieBase = kieContainer.newKieBase(kieBaseConf);

Reply:
Done, but I also changed the others to say "this system property" instead of "this property" for consistency.

And this is the new segment on the exception handler:

```
drools.consequenceExceptionHandler::
When configured, this system property defines the class that manages the exceptions thrown by rule consequences. You can use this property to specify a custom exception handler for rule evaluation in the {DECISION_ENGINE}.
+
--
Default value: `org.drools.core.runtime.rule.impl.DefaultConsequenceExceptionHandler`

You can specify the custom exception handler using one of the following options:

* Specify the exception handler in a system property:
+
[source]
----
drools.consequenceExceptionHandler=org.drools.core.runtime.rule.impl.MyCustomConsequenceExceptionHandler
----

* Specify the exception handler while creating the KIE base programatically:
+
[source,java]
----
KieServices ks = KieServices.Factory.get();
KieBaseConfiguration kieBaseConf = ks.newKieBaseConfiguration(); kieBaseConf.setOption(ConsequenceExceptionHandlerOption.get(MyCustomConsequenceExceptionHandler.class));
KieBase kieBase = kieContainer.newKieBase(kieBaseConf);
----
--
```